languages/java/oso/src/main/java/com/osohq/oso/Query.java

```diff
@@ -76,7 +118,23 @@ private void handleCall(
       ffiQuery.callResult(callId, null);
       return;
     }
+  }
+
+  /** Helper for `NextExternal` query events */
+  private void handleNextExternal(long callId, JSONObject term) throws Exceptions.OsoException {
+    if (!calls.containsKey(callId)) {
+      Object result = host.toJava(term);
```

Comment:
Isn't it possible that `term` is an instance, in which case we would need to use `host.getInstance()`, not `host.toJava()` (the same way we do on lines 70-80?